src/otx/core/model/entity/base.py

```diff
@@ -170,41 +173,67 @@ def map_class_names(src_classes: list[str], dst_classes: list[str]) -> list[int]
+        output_dir: Path,
+        export_format: OTXExportFormatType,
+        precision: OTXExportPrecisionType = OTXExportPrecisionType.FP32,
+    ) -> Path:
         """Export this model to the specified output directory.
 
         Args:
-            output_dir: Directory path to save exported binary files.
-            export_format: Format in which this `OTXModel` is exported.
+            output_dir (Path): directory for saving the exported model
+            export_format (OTXExportFormatType): format of the output model
+            precision (OTXExportPrecisionType): precision of the output model
+        Returns:
+            Path: path to the exported model.
         """
-        if export_format == OTXExportFormat.OPENVINO:
-            self._export_to_openvino(output_dir)
-        if export_format == OTXExportFormat.ONNX:
-            self._export_to_onnx()
-        if export_format == OTXExportFormat.EXPORTABLE_CODE:
-            self._export_to_exportable_code()
+        exporter = self._create_exporter()
+        metadata = self._generate_model_metadata()
 
-    def _export_to_openvino(self, output_dir: Path) -> None:
-        """Export to OpenVINO Intermediate Representation format.
+        if export_format == OTXExportFormatType.OPENVINO:
+            return exporter.to_openvino(self.model, output_dir, self._EXPORTED_MODEL_BASE_NAME, precision, metadata)
+        if export_format == OTXExportFormatType.ONNX:
+            return exporter.to_onnx(self.model, output_dir, self._EXPORTED_MODEL_BASE_NAME, precision, metadata)
+        if export_format == OTXExportFormatType.EXPORTABLE_CODE:
+            return self._export_to_exportable_code()
 
-        Args:
-            output_dir: Directory path to save exported binary files
-        """
+        msg = f"Unsupported export format: {export_format}"
+        raise ValueError(msg)
+
+    def _create_exporter(
+        self,
+    ) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
         raise NotImplementedError
 
-    def _export_to_onnx(self) -> None:
-        """Export to ONNX format.
+    def _generate_model_metadata(
+        self,
+    ) -> dict[tuple[str, str], str]:
+        """Generates model-specific metadata, which will be embedded into exported model.
 
-        Args:
-            output_dir: Directory path to save exported binary files
+        Returns:
+            dict[tuple[str, str], str]: metadata
         """
-        raise NotImplementedError
-
-    def _export_to_exportable_code(self) -> None:
+        all_labels = ""
+        all_label_ids = ""
+        for lbl in self.label_info.label_names:
+            all_labels += lbl.replace(" ", "_") + " "
+            all_label_ids += lbl.replace(" ", "_") + " "
+
+        return {
+            ("model_info", "labels"): all_labels.strip(),
+            ("model_info", "label_ids"): all_label_ids.strip(),
+        }
+
+    def _export_to_exportable_code(self) -> Path:
```

Comment:
Why don't we move this to `exporter` as well.
```suggestion
    exporter.to_exportable_code(...)
```

Reply:
It is done in my PR with exportable code. It will be opened after this one merged.